master/buildbot/changes/gitpoller.py

```diff
@@ -281,9 +282,24 @@ def _removeHeads(self, branch):
             branch = branch[11:]
         return branch
 
-    def _trackerBranch(self, branch):
-        url = urlquote(self.repourl, '').replace('~', '%7E')
-        return f"refs/buildbot/{url}/{self._trim_prefix(branch, 'refs/')}"
+    def _tracker_branch(self, branch: str) -> str:
```

Comment:
I think we should have more tests for this function if we're changing it, especially the case with password which this PR fixes.

Reply:
The _tracker_branch function could be made staticmethod so that whole class instance does not need to be instantiated.